### PR TITLE
Updated RethrowCaughtException according to #906

### PR DESCRIPTION
**Pull request by schalkms**
This implements the rule to be compliant with Sonars rule specification.
The specification can be found here:
https://sbforge.org/sonar/rules/show/squid:S1166?modal=true&layout=false

### CODE REVIEW

File: detekt-rules/src/test/resources/cases/RethrowCaughtExceptionNegative.kt [hunk 1]
```diff
@@ -8,11 +8,11 @@ fun rethrowCaughtExceptionNegative() {
 	}
 	try {
 	} catch (e: IllegalStateException) {
-		print(e) // logging an exception is allowed
+		print(e) // just logging the exception is fine
 	}
 	try {
 	} catch (e: IllegalStateException) {
-		print(e) // logging an exception is allowed
+		print("log") // letting an exception propagate without doing anything with it is fine
```
Comment (marianosimone) [inline detekt-rules/src/test/resources/cases/RethrowCaughtExceptionNegative.kt]: I'd say this is not "doing anything with it". The comment would probably be clearer if it said something:

> letting an exception propagate after doing something with it is fine

Comment (schalkms) [reply detekt-rules/src/test/resources/cases/RethrowCaughtExceptionNegative.kt]: Yes, thanks. I need to update the rule documentation anyway :)

**Comment by arturbosch** [comment]
Unfortunately this case is not supported:
```
// Noncompliant - exception is lost
try { /* ... */ } catch (Exception e) { throw new RuntimeException("context"); }
```

**Comment by arturbosch** [comment]
Please also add this case to the negative file:
```
try { /* ... */ } catch (Exception e) { throw new RuntimeException("context", e); }
```
Its basically like the rethrow case but with one more parameter

**Comment by schalkms** [comment]
> Unfortunately this case is not supported:
```
// Noncompliant - exception is lost
try { /* ... */ } catch (Exception e) { throw new RuntimeException("context"); }
```

This is handled by the `SwallowedException` rule. I didn't want to duplicate the checks.

**Comment by arturbosch** [comment]
Okay, fine by me :)

**Pull request merged**